lib/jeos.pm

```diff
@@ -0,0 +1,60 @@
+# Copyright (C) 2015-2020 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+package jeos;
+use strict;
+use warnings;
+use parent qw(Exporter);
+use testapi;
+use utils qw(ensure_serialdev_permissions);
+use power_action_utils qw(power_action);
+use Utils::Backends qw(is_hyperv);
+use version_utils qw(is_sle is_tumbleweed is_leap);
+use bootloader_setup qw(change_grub_config grep_grub_settings grub_mkconfig set_framebuffer_resolution set_extrabootparams_grub_conf);
+
+our @EXPORT = qw(expect_mount_by_uuid set_grub_gfxmode rebooting is_efi_boot);
+
+sub expect_mount_by_uuid {
+    return (is_hyperv || is_sle('>=15-sp2') || is_tumbleweed || is_leap('>=15.2'));
+}
+
+# non empty */sys/firmware/efi/* must exist in UEFI mode
+sub is_efi_boot {
+    return !script_run('test -d /sys/firmware/efi/') && script_run('test -z "$(ls -A /sys/firmware/efi/)"');
+}
+
+sub rebooting {
+    my ($self, $msg) = @_;
+    power_action('reboot', textmode => 1);
+    record_info('reboot', $msg);
+    wait_serial '/^BdsDxe:\s+failed.*Access\sDenied/', expect_not_found => 1 or
+      record_info('UEFI', 'Access denied reported by MOK', result => 'fail');
+    $self->wait_boot;
+    $self->select_serial_terminal;
+    ensure_serialdev_permissions;
+}
+
+# Set GRUB_GFXMODE to 1024x768
+sub set_grub_gfxmode {
```

Comment:
Not really related to this PR, but I wonder why we need this specifically for JeOS.

Reply:
I assume grub resolution does not bother them. 
https://openqa.suse.de/tests/4146546#step/boot_to_desktop/1 and they set video at the beginning of each install job.

Reply:
The image won't change resolution after it leaves grub2 without updated gfx.
Tested locally and basically that is the same as we do in grub2 when the at the image's first boot. 

https://github.com/os-autoinst/os-autoinst-distri-opensuse/commit/eb635250aacc75e880f7fd7e22db518cb12a6a93